### PR TITLE
adding sync batch to torchsparse++

### PR DESCRIPTION
**Pull request by rui-wang-ryan**
This PR unblocks sync batchnorm usage in torch sparse sparse, such that the statistic from batches of all GPUs are synced.

The detail objective and training experiment can be found at [sync batch experiment](https://zendar.atlassian.net/browse/RP-2461).

FYI, I have created a test case, not sure if this repo run any test. Probably put it in radar processor then.

@tian-zhou-io can you be the main reviewer?
CC @kellman

### CODE REVIEW

**Review by tian-zhou-io — APPROVED** [review]
Looks good to me! Just to confirm — is setting the flag to True the only change needed in the training script? Also, when do you plan to update the RP with the flag change and the test?

**Comment by rui-wang-ryan** [comment]
> Looks good to me! Just to confirm — is setting the flag to True the only change needed in the training script? Also, when do you plan to update the RP with the flag change and the test?

I will create a separate PR to apply this to multi-gpu train script.

**Comment by rui-wang-ryan** [comment]
after discussing with @Ajaydip Due to this CI issue, and urgency of training new  model and small code changes, will make the code changes in radar processor and close this PR.
